public/pages/Channels/Channels.tsx

```diff
@@ -116,14 +119,18 @@ export class Channels extends Component<ChannelsProps, ChannelsState> {
   }
 
   async componentDidUpdate(prevProps: ChannelsProps, prevState: ChannelsState) {
-    const prevQuery = Channels.getQueryObjectFromState(prevState);
-    const currQuery = Channels.getQueryObjectFromState(this.state);
+    const prevQuery = this.getQueryObjectFromState(prevState);
+    const currQuery = this.getQueryObjectFromState(this.state);
+
     if (!_.isEqual(prevQuery, currQuery)) {
       await this.refresh();
     }
+    if (isDataSourceChanged(this.props, prevProps)) {
```

Comment:
Nitpick:
This check could be worked into line check on line 125 by changing the check to
```
if (!_.isEqual(prevQuery, currQuery) || isDataSourceChanged(this.props, prevProps)) { ...